### PR TITLE
Order details page

### LINKED ISSUE #260: Order details view
Implement an order details view. This view will be available either from the Order History view (#259) or from the order confirmation page (#252). 

Data required to render the view as it looks on mockups:

```graphql
{
  order(id: $id) {
    id
    lines {
      id
      productName
      thumbnailUrl
      unitPrice {
        gross {
          localized
        }
      }
      quantity
    }
    subtotal {
      gross {
        localized
      }
    }
    total {
      gross {
        localized
      }
    }
    shippingPrice {
      gross {
        localized
      }
    }
    shippingAddress {
      ...Addresss
    }
  }
}
```

API doesn't yet provide data for:
- contact information
- estimated delivery

### PR DESCRIPTION
**Pull request by piotrgrundas**
Resolves #260 

Implemented essentially two the same views
`/order/:token/` for guest users
`/my-account/order/:id/` for logged in users

image
image
image

### CODE REVIEW

File: src/components/AddressSummary/index.tsx [hunk 1]
```diff
@@ -1,10 +1,16 @@
...
+import { getCheckout_checkout_shippingAddress } from "../../checkout/types/getCheckout";
+import { OrderById_order_shippingAddress } from "../../userAccount/views/OrderDetails/types/OrderById";
+import { OrderByToken_orderByToken_shippingAddress } from "../../userAccount/views/OrderDetails/types/OrderByToken";
 
 const AddressSummary: React.FC<{
-  address: getCheckout_checkout_shippingAddress;
+  address:
+    | getCheckout_checkout_shippingAddress
+    | OrderById_order_shippingAddress
+    | OrderByToken_orderByToken_shippingAddress;
```
Comment (dominik-zeglen) [inline src/components/AddressSummary/index.tsx]: Can't we just type it as a fragment?

File: src/components/CartTable/ProductRow.tsx [hunk 2]
```diff
@@ -16,9 +18,12 @@ import cartSubtractImg from "../../images/cart-subtract.svg";
 
 export type LineI = (
   | VariantList_productVariants_edges_node
-  | Checkout_lines_variant) & {
+  | Checkout_lines_variant
```
Comment (dominik-zeglen) [inline src/components/CartTable/ProductRow.tsx]: Again, fragment maybe?

File: src/userAccount/views/OrderDetails/Page.tsx [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+  order: OrderById_order | OrderByToken_orderByToken;
+}> = ({ guest, order }) =>
+  order ? (
+    <>
+      {!guest && (
+        <Link className="order-details__link" to={orderHistoryUrl}>
+          Go back to Order History
+        </Link>
+      )}
+      <h3>Your order nr: {order.number}</h3>
```
Comment (NyanKiyoshi) [inline src/userAccount/views/OrderDetails/Page.tsx]: It looks weird to me. The format `Order #{order_number}` wouldn't be better looking?

Comment (mridang) [reply src/userAccount/views/OrderDetails/Page.tsx]: I agree.

**Comment by mridang** [comment]
I wasn't able to get this screen to work. I pulled the latest master for the `saleor` backend and ran this pull-request against it. When trying to view the details of an order, I get an error.

` TS2339: Property 'variant' does not exist on type 'OrderById_order_lines | OrderByToken_orderByToken_lines'.
  Property 'variant' does not exist on type 'OrderById_order_lines'.`

I ran this GraphQL query against the backed and got the same error.

```
fragment OrderPrice on TaxedMoney {
  gross {
    localized
    __typename
  }
  __typename
}

fragment Address on Address {
  firstName
  lastName
  companyName
  streetAddress1
  streetAddress2
  city
  postalCode
  country {
    code
    country
    __typename
  }
  countryArea
  phone
  __typename
}


fragment OrderDetail on Order {
  userEmail
  paymentStatus
  status
  id
  number
  shippingAddress {
    ...Address
    __typename
  }
  lines {
    productName
    quantity
    id
		unitPrice {
      gross {
        amount
      }
      currency
    }
    quantity
    __typename
  }
  subtotal {
    ...OrderPrice
    __typename
  }
  total {
    ...OrderPrice
    __typename
  }
  shippingPrice {
    ...OrderPrice
    __typename
  }
  __typename
}

query OrderByToken($token: String!) {
  orderByToken(token: $token) {
    ...OrderDetail
    __typename
  }
}```

Is there something that needs to be done in order to get this working or should I report this a bug? Thank you.

**Pull request merged**